### PR TITLE
Replace DP/DDP/DDPSpawn plugins to strategies, keep the old for compatibility

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

Error:

```bash
File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/flash/image/classification/adapters.py", line 23, in <module>
    from pytorch_lightning.plugins import DataParallelPlugin, DDPPlugin, DDPSpawnPlugin
ImportError: cannot import name 'DataParallelPlugin' from 'pytorch_lightning.plugins' 
```

PL 1.6.0 had removed these plugins, in the favor of strategies. This PR makes sure to not break the compatibility for older PL versions, and making sure to support latest PL API.

## Before submitting
- [ ] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

File: flash/image/classification/adapters.py [hunk 1]
```diff
@@ -242,13 +246,23 @@ def _convert_dataset(
...
-                DDPPlugin,
-                DDPSpawnPlugin,
-            ),
-        ):
+        if _PL_GREATER_EQUAL_1_6_0:
+            is_ddp_or_ddp_spawn = isinstance(
+                trainer.strategy,
+                (
+                    DDPStrategy,
+                    DDPSpawnStrategy,
```
Comment (Borda) [inline flash/image/classification/adapters.py:254]:
```suggestion
                    DDPSpawnStrategy
```

File: flash/image/classification/adapters.py [hunk 2]
```diff
@@ -242,13 +246,23 @@ def _convert_dataset(
...
+                ),
+            )
+        else:
+            is_ddp_or_ddp_spawn = isinstance(
+                trainer.training_type_plugin,
+                (
+                    DDPPlugin,
+                    DDPSpawnPlugin,
```
Comment (Borda) [inline flash/image/classification/adapters.py:262]:
```suggestion
                    DDPSpawnPlugin
```

**Review by Borda — APPROVED** [review]

**Comment by krshrimali** [comment]
The failing test is flaky (and not related to this PR), from an offline discussion with @ethanwharris:

> We would just need to use a different random seed where it performs better.

For now, merging this PR. 🔥

**Pull request merged**